salt/netapi/rest_cherrypy/app.py

```diff
@@ -953,11 +962,11 @@ def urlencoded_processor(entity):
 
     :param entity: raw POST data
     '''
-    # First call out to CherryPy's default processor
-    cherrypy._cpreqbody.process_urlencoded(entity)
-    cherrypy._cpreqbody.process_urlencoded(entity)
-    cherrypy.serving.request.unserialized_data = entity.params
-    cherrypy.serving.request.raw_body = ''
+    # cherrypy._cpreqbody.process_urlencoded doesn't preserve the raw
+    # "body", so we have to handle parsing the tokens using parse_qsl
+    urlencoded = str(entity.read())
```

Comment:
Why do we need to cast this as `str`?

Reply:
I don't really recall at this point, but I seem to remember that I hit a weird edge case during testing. Problem?